torch/csrc/api/src/nn/init.cpp

```diff
@@ -0,0 +1,67 @@
+#include <torch/nn/init.h>
+
+#include <torch/tensor.h>
+#include <torch/utils.h>
+
+namespace torch {
+namespace nn {
+namespace init {
+
+Tensor uniform_(Tensor tensor, double low, double high) {
+  NoGradGuard guard;
+  return tensor.uniform_(low, high);
+}
+
+Tensor normal_(Tensor tensor, double mean, double std) {
+  NoGradGuard guard;
+  return tensor.normal_(mean, std);
+}
+
+Tensor constant_(Tensor tensor, Scalar value) {
+  NoGradGuard guard;
+  return tensor.fill_(value);
+}
+
+Tensor ones_(Tensor tensor) {
+  NoGradGuard guard;
+  return tensor.fill_(1);
+}
+
+Tensor zeros_(Tensor tensor) {
+  NoGradGuard guard;
+  return tensor.zero_();
+}
+
+Tensor eye_(Tensor tensor) {
+  NoGradGuard guard;
+  return tensor.eye_();
+}
+
+Tensor dirac_(Tensor tensor) {
+  NoGradGuard guard;
+  return tensor.dirac_();
+}
+
+Tensor xavier_uniform_(Tensor tensor, double gain) {
+  NoGradGuard guard;
+  return tensor.xavier_uniform_(gain);
+}
+
+Tensor xavier_normal_(Tensor tensor, double gain) {
+  NoGradGuard guard;
+  return tensor.xavier_normal_(gain);
+}
+
+Tensor orthogonal_(Tensor tensor, double gain) {
+  NoGradGuard guard;
+  return tensor.orthogonal_(gain);
+}
+
+Tensor sparse_(Tensor tensor, double sparsity, double std) {
+  NoGradGuard guard;
+  return tensor.sparse_(sparsity, std);
```

Comment:
We have already discussed this in the past, but it annoys me that we are leaking so many methods to the tensor namespace.
This one is particularly confusing I'd say, as it could also mean that we convert the tensor in place to a sparse tensor

Reply:
Should we put them only on torch and not tensor? We only need them for nn.init, so happy to do whatever is the best way to hide it

Reply:
Yeah, I think it would be better to move the implementations into this file, and maybe remove them from ATen? Not sure what the others think

Reply:
How about adding a namespace argument to native_functions.yaml (for nn, nn.init or so)?

Reply:
@fmassa I do think it's worth having ATen<->Python bindings in order to reduce code duplication, since some of the methods here, like `orthogonal`, are not so short.

@t-vi we could definitely make that change, but I think we'd rather not increase the number of features of our codegen at the moment as it makes it harder to design future libraries like c10

In the past we solved this problem by prefixing such implementation methods with an underscore. I don't find that very clean either though l